src/bp-messages/classes/class-bp-messages-message.php

```diff
@@ -301,13 +301,16 @@ public static function get_last_sent_for_user( $thread_id ) {
 	 *                  sender, otherwise null.
 	 */
 	public static function is_user_sender( $user_id, $message_id ) {
-		global $wpdb;
-
-		$bp = buddypress();
 
-		$query = $wpdb->get_var( $wpdb->prepare( "SELECT id FROM {$bp->messages->table_name_messages} WHERE sender_id = %d AND id = %d", $user_id, $message_id ) );
+		$query = self::get(
+			array(
+				'fields'  => 'ids',
+				'user_id' => $user_id,
+				'include' => array( $message_id ),
+			)
+		);
 
-		return is_numeric( $query ) ? (int) $query : $query;
+		return is_numeric( $query['messages'][0] ) ? (int) $query['messages'][0] : $query['messages'][0];
```

Comment:
@janakkaneriya007 Can you check $wpdb->get_var() function which value return is result is empty. we need to return same value in else part.